### PR TITLE
[FLINK-19119][python][docs] Update the documentation to use Expression instead of strings in the Python Table API

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request updates the documentation to use Expression instead of strings in the Python Table API.*

## Brief change log

  - *Update the documentation to use Expression instead of strings in the Python Table API*

## Verifying this change

This change is a documentation rework without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: docs/dev/python/table-api-users-guide/udfs/python_udfs.zh.md [hunk 1]
```diff
@@ -201,11 +209,11 @@ my_table = ...  # type: Table, table schema: [a: String]
...
 
 # 注册java自定义函数。
-table_env.register_java_function("split", "my.java.function.Split")
+table_env.create_java_temporary_function("split", "my.java.function.Split")
 
 # 在Python Table API中使用表值函数。 "as"指定表的字段名称。
-my_table.join_lateral("split(a) as (word, length)").select("a, word, length")
-my_table.left_outer_join_lateral("split(a) as (word, length)").select("a, word, length")
+my_table.join_lateral(call('split', my_table.a).alias("word, length")).select("a, word, length")
+my_table.left_outer_join_lateral(call('split', my_table.a).alias("word, length")).select("a, word, length")
```
Comment (WeiZhong94) [inline docs/dev/python/table-api-users-guide/udfs/python_udfs.zh.md:216]: ditto

File: docs/dev/python/table-api-users-guide/conversion_of_pandas.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ import numpy as np
 
 # Create a PyFlink Table
 pdf = pd.DataFrame(np.random.rand(1000, 2))
-table = t_env.from_pandas(pdf, ["a", "b"]).filter("a > 0.5")
+table = t_env.from_pandas(pdf, ["a", "b"]).filter(col('a' > 0.5))
```
Comment (WeiZhong94) [inline docs/dev/python/table-api-users-guide/conversion_of_pandas.md:77]: col('a' > 0.5) -> col('a') > 0.5 ?

File: docs/dev/python/table-api-users-guide/conversion_of_pandas.zh.md [hunk 3]
```diff
@@ -71,7 +71,7 @@ import numpy as np
 
 # 创建PyFlink Table
 pdf = pd.DataFrame(np.random.rand(1000, 2))
-table = t_env.from_pandas(pdf, ["a", "b"]).filter("a > 0.5")
+table = t_env.from_pandas(pdf, ["a", "b"]).filter(col('a' > 0.5))
```
Comment (WeiZhong94) [inline docs/dev/python/table-api-users-guide/conversion_of_pandas.zh.md:74]: ditto

File: docs/dev/python/table-api-users-guide/udfs/python_udfs.md [hunk 4]
```diff
@@ -206,11 +214,11 @@ my_table = ...  # type: Table, table schema: [a: String]
 table_env.get_config().get_configuration().set_string("taskmanager.memory.task.off-heap.size", '80m')
 
 # Register the java function.
-table_env.register_java_function("split", "my.java.function.Split")
+table_env.create_java_temporary_function("split", "my.java.function.Split")
 
 # Use the table function in the Python Table API. "as" specifies the field names of the table.
```
Comment (WeiZhong94) [inline docs/dev/python/table-api-users-guide/udfs/python_udfs.md:219]: "as" -> "alias" ?

File: docs/dev/table/tableApi.md [hunk 5]
```diff
@@ -537,11 +539,13 @@ root
...
 orders = t_env.from_path("Orders")
-result = orders.select("a, c as d")
+result = orders.select(orders.a, orders.c.alias('d'))
 {% endhighlight %}
         <p>You can use star (<code>*</code>) to act as a wild card, selecting all of the columns in the table.</p>
 {% highlight python %}
-result = orders.select("*")
+from pyflink.table.expressions import lit
+
+result = orders.select(lit("*"))
```
Comment (WeiZhong94) [inline docs/dev/table/tableApi.md:548]: lit("*") -> col("*") ?

File: docs/dev/table/tableApi.md [hunk 6]
```diff
@@ -1095,11 +1103,12 @@ result = orders.group_by("a").select("a, b.sum as d")
...
 {% highlight python %}
 from pyflink.table.window import Tumble
+from pyflink.table.expressions import lit, col
 
 orders = t_env.from_path("Orders")
-result = orders.window(Tumble.over("5.minutes").on("rowtime").alias("w")) \ 
-               .group_by("a, w") \
-               .select("a, w.start, w.end, b.sum as d")
+result = orders.window(Tumble.over(lit(5).minutes).on(orders.rowtime).alias("w")) \ 
+               .group_by(col('a'), col('w')) \
```
Comment (WeiZhong94) [inline docs/dev/table/tableApi.md:1110]: col('a') -> orders.a ?

File: docs/dev/table/tableApi.md [hunk 7]
```diff
@@ -1131,22 +1141,23 @@ result = orders.over_window(Over.partition_by("a").order_by("rowtime")
...
 # Distinct aggregation on group by
-group_by_distinct_result = orders.group_by("a") \
-                                 .select("a, b.sum.distinct as d")
+group_by_distinct_result = orders.group_by(orders.a) \
+                                 .select(orders.a, orders.b.sum.distinct.alias('d'))
 # Distinct aggregation on time window group by
 group_by_window_distinct_result = orders.window(
-    Tumble.over("5.minutes").on("rowtime").alias("w")).group_by("a, w") \
-    .select("a, b.sum.distinct as d")
+    Tumble.over(lit(5).minutes).on(orders.rowtime).alias("w")).group_by(col('a'), col('w')) \
```
Comment (WeiZhong94) [inline docs/dev/table/tableApi.md:1152]: col('a') -> orders.a ?

File: docs/dev/table/tableApi.md [hunk 8]
```diff
@@ -1494,9 +1505,11 @@ val result = orders
...
         <p>Similar to a SQL JOIN clause. Joins two tables. Both tables must have distinct field names and at least one equality join predicate must be defined through join operator or using a where or filter operator.</p>
 {% highlight python %}
-left = t_env.from_path("Source1").select("a, b, c")
-right = t_env.from_path("Source2").select("d, e, f")
-result = left.join(right).where("a = d").select("a, b, e")
+from pyflink.table.expressions import col
+
+left = t_env.from_path("Source1").select(col('a'), col('b'), col('c'))
+right = t_env.from_path("Source2").select(col('d'), col('e'), col('f'))
+result = left.join(right).where(left.a == right.d).select(col('a'), col('b'), col('e'))
```
Comment (WeiZhong94) [inline docs/dev/table/tableApi.md:1512]: select(col('a'), col('b'), col('e')) -> select(left.a, left.b, right.e) ?

File: docs/dev/table/tableApi.md [hunk 9]
```diff
@@ -1961,15 +1982,15 @@ result = left.minus_all(right)
...
-left = t_env.from_path("Source1").select("a, b, c")
-right = t_env.from_path("Source2").select("a")
+left = t_env.from_path("Source1").select(col('a'), col('b'), col('c'))
+right = t_env.from_path("Source2").select(col('a'))
 
 # using implicit registration
-result = left.select("a, b, c").where("a.in(%s)" % right)
+result = left.select(col('a'), col('b'), col('c')).where(col('a').in_(right))
 
 # using explicit registration
```
Comment (WeiZhong94) [inline docs/dev/table/tableApi.md:1991]: The implicit/explicit registration could be removed as Java has removed them. We only need to show the expression usage.

**Review by WeiZhong94 — COMMENTED** [review]
@dianfu Thanks for your PR! I have left some comments.

File: docs/dev/python/table_api_tutorial.md [hunk 10]
```diff
@@ -124,10 +124,12 @@ The table `mySink` has two columns, word and count, and writes data to the file
...
 {% highlight python %}
-t_env.from_path('mySource') \
-    .group_by('word') \
-    .select('word, count(1)') \
-    .insert_into('mySink')
+from pyflink.table.expressions import lit
+
+tab = t_env.from_path('mySource')
+tab.group_by(tab.word) \
+   .select(tab.word, lit(1).count) \
```
[the remark this answers is not in the document]
Comment (dianfu) [reply docs/dev/python/table_api_tutorial.md:131]: I think there is no need to use 4 spaces here.

File: docs/dev/table/tableApi.md [hunk 11]
```diff
@@ -1562,11 +1579,12 @@ result = left.join(right).where("a = d && rowtime1 >= rowtime2 - 1.second
 @udtf(result_types=[DataTypes.BIGINT(), DataTypes.BIGINT(), DataTypes.BIGINT()])
 def split(x):
     return [Row(1, 2, 3)]
-t_env.register_function("split", split)
+t_env.create_temporary_function("split", split)
```
Comment (WeiZhong94) [inline docs/dev/table/tableApi.md:1582]: It seems we can remove the registration.

File: docs/dev/table/tableApi.md [hunk 12]
```diff
@@ -1583,11 +1601,12 @@ result = orders.join_lateral("split(c).as(s, t, v)").select("a, b, s, t, v")
...
```
Comment (WeiZhong94) [inline docs/dev/table/tableApi.md:1604]: It seems we can remove the registration.

**Review by WeiZhong94 — COMMENTED** [review]
@dianfu Thanks for your update. Looks good overall, just some minor comments.

**Review by WeiZhong94 — APPROVED** [review]

**Comment by dianfu** [comment]
@WeiZhong94 Thanks a lot for the review. I will integrate the last two comments during merging.

**Pull request closed without merging**